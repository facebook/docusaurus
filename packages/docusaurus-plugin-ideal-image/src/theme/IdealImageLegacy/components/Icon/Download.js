// This icon is from "material design icons"
// It is licensed under Apache License 2.0
// Full text is available here
// https://github.com/google/material-design-icons/blob/master/LICENSE
import React from 'react';
import Icon from './index';

const Download = (props) => (
  <Icon
    {...props}
    path="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
  />
);

export default Download;
