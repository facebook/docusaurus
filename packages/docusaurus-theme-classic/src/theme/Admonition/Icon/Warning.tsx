/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import type {Props} from '@theme/Admonition/Icon/Warning';

export default function AdmonitionIconCaution(props: Props): JSX.Element {
  return (
    <svg viewBox="0 0 16 16" {...props}>
      <path
        fillRule="evenodd"
        d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"
      />
    </svg>
  );
}
