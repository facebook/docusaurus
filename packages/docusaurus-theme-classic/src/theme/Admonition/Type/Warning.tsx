/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import clsx from 'clsx';
import Translate from '@docusaurus/Translate';
import type {Props} from '@theme/Admonition/Type/Warning';
import AdmonitionLayout from '@theme/Admonition/Layout';
import IconWarning from '@theme/Admonition/Icon/Warning';

const infimaClassName = 'alert alert--warning';

const defaultProps = {
  icon: <IconWarning />,
  title: (
    <Translate
      id="theme.admonition.warning"
      description="The default label used for the Warning admonition (:::warning)">
      warning
    </Translate>
  ),
};

export default function AdmonitionTypeWarning(props: Props): JSX.Element {
  return (
    <AdmonitionLayout
      {...defaultProps}
      {...props}
      className={clsx(infimaClassName, props.className)}>
      {props.children}
    </AdmonitionLayout>
  );
}
