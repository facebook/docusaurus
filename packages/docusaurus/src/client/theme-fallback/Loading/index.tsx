/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

// Should we translate theme-fallback?
/* eslint-disable @docusaurus/no-untranslated-text */

import React from 'react';
import type {LoadingComponentProps} from 'react-loadable';

export default function Loading({
  error,
  retry,
  pastDelay,
}: LoadingComponentProps): JSX.Element | null {
  if (error) {
    return (
      <div
        style={{
          textAlign: 'center',
          color: '#fff',
          backgroundColor: '#fa383e',
          borderColor: '#fa383e',
          borderStyle: 'solid',
          borderRadius: '0.25rem',
          borderWidth: '1px',
          boxSizing: 'border-box',
          display: 'block',
          padding: '1rem',
          flex: '0 0 50%',
          marginLeft: '25%',
          marginRight: '25%',
          marginTop: '5rem',
          maxWidth: '50%',
          width: '100%',
        }}>
        <p>{String(error)}</p>
        <div>
          <button type="button" onClick={retry}>
            Retry
          </button>
        </div>
      </div>
    );
  }

  if (pastDelay) {
    return (
      <div
        style={{
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          height: '100vh',
        }}>
        <svg
          id="loader"
          style={{
            width: 128,
            height: 110,
            position: 'absolute',
            top: 'calc(100vh - 64%)',
          }}
          viewBox="0 0 45 45"
          xmlns="http://www.w3.org/2000/svg"
          stroke="#61dafb">
          <g
            fill="none"
            fillRule="evenodd"
            transform="translate(1 1)"
            strokeWidth="2">
            <circle cx="22" cy="22" r="6" strokeOpacity="0">
              <animate
                attributeName="r"
                begin="1.5s"
                dur="3s"
                values="6;22"
                calcMode="linear"
                repeatCount="indefinite"
              />
              <animate
                attributeName="stroke-opacity"
                begin="1.5s"
                dur="3s"
                values="1;0"
                calcMode="linear"
                repeatCount="indefinite"
              />
              <animate
                attributeName="stroke-width"
                begin="1.5s"
                dur="3s"
                values="2;0"
                calcMode="linear"
                repeatCount="indefinite"
              />
            </circle>
            <circle cx="22" cy="22" r="6" strokeOpacity="0">
              <animate
                attributeName="r"
                begin="3s"
                dur="3s"
                values="6;22"
                calcMode="linear"
                repeatCount="indefinite"
              />
              <animate
                attributeName="stroke-opacity"
                begin="3s"
                dur="3s"
                values="1;0"
                calcMode="linear"
                repeatCount="indefinite"
              />
              <animate
                attributeName="stroke-width"
                begin="3s"
                dur="3s"
                values="2;0"
                calcMode="linear"
                repeatCount="indefinite"
              />
            </circle>
            <circle cx="22" cy="22" r="8">
              <animate
                attributeName="r"
                begin="0s"
                dur="1.5s"
                values="6;1;2;3;4;5;6"
                calcMode="linear"
                repeatCount="indefinite"
              />
            </circle>
          </g>
        </svg>
      </div>
    );
  }

  return null;
}
