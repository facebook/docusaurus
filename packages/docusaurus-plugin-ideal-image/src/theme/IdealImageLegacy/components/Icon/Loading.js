// This icon is from "material design icons"
// It is licensed under Apache License 2.0
// Full text is available here
// https://github.com/google/material-design-icons/blob/master/LICENSE
import React from 'react';
import Icon from './index';

const Loading = (props) => (
  <Icon
    {...props}
    path="M6,2V8H6V8L10,12L6,16V16H6V22H18V16H18V16L14,12L18,8V8H18V2H6M16,16.5V20H8V16.5L12,12.5L16,16.5M12,11.5L8,7.5V4H16V7.5L12,11.5Z"
  />
);

export default Loading;
