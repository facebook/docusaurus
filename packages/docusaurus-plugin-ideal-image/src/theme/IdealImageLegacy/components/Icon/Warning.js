// This icon is from "material design icons"
// It is licensed under Apache License 2.0
// Full text is available here
// https://github.com/google/material-design-icons/blob/master/LICENSE
import React from 'react';
import Icon from './index';

const Warning = (props) => (
  <Icon
    {...props}
    path="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
  />
);

export default Warning;
