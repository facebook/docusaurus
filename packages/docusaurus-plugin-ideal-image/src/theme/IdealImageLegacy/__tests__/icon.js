import React from 'react';
import renderer from 'react-test-renderer';
import Download from '../components/Icon/Download';
import Loading from '../components/Icon/Loading';
import Offline from '../components/Icon/Offline';
import Warning from '../components/Icon/Warning';

const snapshotTestDescription = 'Should render a snapshot that is good';

describe('Download icon', () => {
  it(snapshotTestDescription, () => {
    const download = renderer.create(<Download />).toJSON();
    expect(download).toMatchSnapshot();
  });
});

describe('Loading icon', () => {
  it(snapshotTestDescription, () => {
    const loading = renderer.create(<Loading />).toJSON();
    expect(loading).toMatchSnapshot();
  });
});

describe('Offline icon', () => {
  it(snapshotTestDescription, () => {
    const offline = renderer.create(<Offline />).toJSON();
    expect(offline).toMatchSnapshot();
  });
});

describe('Warning icon', () => {
  it(snapshotTestDescription, () => {
    const warning = renderer.create(<Warning />).toJSON();
    expect(warning).toMatchSnapshot();
  });
});
