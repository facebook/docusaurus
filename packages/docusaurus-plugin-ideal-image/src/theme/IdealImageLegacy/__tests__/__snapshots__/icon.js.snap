// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download icon Should render a snapshot that is good 1`] = `
<svg
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
    fill="#000"
  />
</svg>
`;

exports[`Loading icon Should render a snapshot that is good 1`] = `
<svg
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M6,2V8H6V8L10,12L6,16V16H6V22H18V16H18V16L14,12L18,8V8H18V2H6M16,16.5V20H8V16.5L12,12.5L16,16.5M12,11.5L8,7.5V4H16V7.5L12,11.5Z"
    fill="#000"
  />
</svg>
`;

exports[`Offline icon Should render a snapshot that is good 1`] = `
<svg
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M19.35 10.04C18.67 6.59 15.64 4 12 4c-1.48 0-2.85.43-4.01 1.17l1.46 1.46C10.21 6.23 11.08 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3 0 1.13-.64 2.11-1.56 2.62l1.45 1.45C23.16 18.16 24 16.68 24 15c0-2.64-2.05-4.78-4.65-4.96zM3 5.27l2.75 2.74C2.56 8.15 0 10.77 0 14c0 3.31 2.69 6 6 6h11.73l2 2L21 20.73 4.27 4 3 5.27zM7.73 10l8 8H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h1.73z"
    fill="#000"
  />
</svg>
`;

exports[`Warning icon Should render a snapshot that is good 1`] = `
<svg
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    fill="#000"
  />
</svg>
`;
